### ISSUE TITLE
RDS: publicly_accessible setting not updating after initial cluster and instance creation

### ISSUE THREAD

**Issue by sripulip** [terraform-providers/terraform-provider-aws#5691]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
$ terraform -v
Terraform v0.11.8
+ provider.aws v1.33.0
```

### Affected Resource(s)
* aws_rds_cluster_instance

### Terraform Configuration Files

```hcl
provider "aws" {
  region = "us-west-2"
}

resource "aws_rds_cluster" "aurora_cluster" {
  cluster_identifier  = "test-publicly-accessible-behavior"
  master_username     = "username"
  master_password     = "password"
  skip_final_snapshot = true
}

resource "aws_rds_cluster_instance" "aurora_instances" {
  count               = "1"
  identifier          = "test-publicly-accessible-behavior-${count.index}"
  cluster_identifier  = "${aws_rds_cluster.aurora_cluster.id}"
  instance_class      = "db.t2.small"
  publicly_accessible = true
  apply_immediately   = true
}
```
and manually modified the database security group inbound rules to allow access from `0.0.0.0/0`.

### Expected Behavior

`Publicly accessible` should have changed from "Yes" to "No" in the AWS console and the database should no longer be accessible over the internet.

### Actual Behavior

`Publicly accessible` setting hasn't changed. Database is still accessible over the internet. `terraform plan` recognizes the change, but nothing actually changes after doing a `terraform apply` even though the console/log message says otherwise.

### Steps to Reproduce

1. `terraform apply`
2. Set `publicly_accessible = false`
3. `terraform apply`

### Important Factoids
Using AWS APIs directly (as shown below) modified the setting as expected:
```
import boto3

client = boto3.client('rds', region_name='us-west-2')

client.modify_db_instance(
    DBInstanceIdentifier='test-publicly-accessible-behavior-0',
    PubliclyAccessible=False
)
```

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 1.38.0 of the AWS provider, likely later today.

**Comment by bflad** [comment]
This has been released in [version 1.38.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1380-september-26-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
